storage/cache/subtree_cache.go

```diff
@@ -424,7 +441,9 @@ func PopulateLogSubtreeNodes(treeHasher merkle.TreeHasher) storage.PopulateSubtr
 					// TODO(al): Don't panic Mr. Mainwaring.
 					panic(err)
 				}
-				if depth > 0 {
+				// Don't put leaves into the internal map and only update if we're rebuilding internal
```

Comment:
Not necessarily. We always have to go through the loop because we need the root hash from the CMT as we don't store it (there's a TODO). In the case where the internal leaves were saved with the subtree we don't update the map because they're correct when we started. I've tried a different explanation.